src/PiGui.cpp

```diff
@@ -220,19 +225,27 @@ bool PiGui::CircularSlider(const ImVec2 &center, float *v, float v_min, float v_
 	return ImGui::SliderBehavior(ImRect(center.x - 17, center.y - 17, center.x + 17, center.y + 17), id, v, v_min, v_max, 1.0, 4);
 }
 
-void *PiGui::makeTexture(unsigned char *pixels, int width, int height) {
-	GLint last_texture;
-	GLuint result;
-	glGetIntegerv(GL_TEXTURE_BINDING_2D, &last_texture);
-	glGenTextures(1, &result);
-	glBindTexture(GL_TEXTURE_2D, result);
-	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-	glPixelStorei(GL_UNPACK_ROW_LENGTH, 0);
-	glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA, width, height, 0, GL_RGBA, GL_UNSIGNED_BYTE, pixels);
-	glBindTexture(GL_TEXTURE_2D, last_texture);
-	Output("texture id: %i\n", result);
-	return reinterpret_cast<void*>(result);
+void *PiGui::makeTexture(unsigned char *pixels, int width, int height)
+{
+	// this is not very pretty code and uses the Graphics::TextureGL class directly
+	// Texture descriptor defines the size, type.
+	// Gone for LINEAR_CLAMP here and RGBA like the original code
+	const vector2f texSize(1.0f, 1.0f);
+	const vector2f dataSize(width, height);
+	const Graphics::TextureDescriptor texDesc(Graphics::TEXTURE_RGBA_8888,
+																						dataSize, texSize, Graphics::LINEAR_CLAMP,
+																						false, false, false, 0, Graphics::TEXTURE_2D);
```

Comment:
...and here? I prefer alidning at the opening ´(´, or one indentation level from the ´cons´